sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -338,10 +354,13 @@ def test_enet_positive_constraint():
     X = [[-1], [0], [1]]
     y = [1, 0, -1]       # just a straight line with negative slope
 
-    enet = ElasticNet(alpha=0.1, max_iter=1000, positive=True)
+    enet = ElasticNet(alpha=0.1, max_iter=1000, positive=True, solver='cd')
     enet.fit(X, y)
     assert min(enet.coef_) >= 0
 
+    enet = ElasticNet(alpha=0.1, max_iter=1000, positive=True, solver='saga')
+    assert_raises(ValueError, enet.fit, X, y)
```

Comment:
we tend to prefer `with pytest.raises(error, match=message)` constructs.

Reply:
Done.